zrml/court/src/tests.rs

```diff
@@ -63,3 +67,43 @@ fn join_court_will_not_insert_an_already_stored_juror() {
         );
     });
 }
+
+#[test]
+fn random_jurors_return_a_subset_of_jurors() {
+    ExtBuilder::default().build().execute_with(|| {
+        const START_BLOCK: u64 = 123;
+        setup_blocks(1..START_BLOCK);
+        let jurors = [
+            (0, Juror { staked: 1, status: JurorStatus::Ok }),
+            (0, Juror { staked: 2, status: JurorStatus::Tardy }),
+            (0, Juror { staked: 3, status: JurorStatus::Ok }),
+            (0, Juror { staked: 4, status: JurorStatus::Tardy }),
+            (0, Juror { staked: 5, status: JurorStatus::Ok }),
+            (0, Juror { staked: 6, status: JurorStatus::Ok }),
+            (0, Juror { staked: 7, status: JurorStatus::Ok }),
+        ];
+        let random_jurors = Court::random_jurors(&jurors, 2);
+        let mut at_least_one_is_different = false;
+        for n in 0..1000 {
+            setup_blocks(START_BLOCK..START_BLOCK + n);
+            if random_jurors != Court::random_jurors(&jurors, 2) {
+                at_least_one_is_different = true;
+                break;
+            }
+        }
```

Comment:
This for loop is unable to detect whether the two arrays contains different jurors, it can only check if the complete arrays differ. If you run this test with a println!(...) command in the current state, that prints both arrays for every iteration, and additionally comment out the break statement, you will get an output like this:
> [Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }] - [Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }]
[Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }] - [Juror { staked: 1, status: Ok }, Juror { staked: 2, status: Tardy }]
[Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }] - [Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }]
[Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }] - [Juror { staked: 1, status: Ok }, Juror { staked: 2, status: Tardy }]
[Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }] - [Juror { staked: 1, status: Ok }, Juror { staked: 2, status: Tardy }]
[Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }] - [Juror { staked: 1, status: Ok }, Juror { staked: 2, status: Tardy }]
[Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }] - [Juror { staked: 1, status: Ok }, Juror { staked: 2, status: Tardy }]
[Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }] - [Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }]
[Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }] - [Juror { staked: 1, status: Ok }, Juror { staked: 2, status: Tardy }]
[Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }] - [Juror { staked: 1, status: Ok }, Juror { staked: 2, status: Tardy }]
[Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }] - [Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }]
[Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }] - [Juror { staked: 2, status: Tardy }, Juror { staked: 1, status: Ok }]

Cleary the test checks whether the arrays differ, but not whether different jurors were selected.
```suggestion
        for n in 0..1000 {
            setup_blocks(START_BLOCK..START_BLOCK + n);
            rand_jurs = Court::random_jurors(&jurors, 2);
            for juror in &random_jurors {
                at_least_one_is_different = rand_jurs.iter().all(|&el| el != *juror);
                if at_least_one_is_different { return };
            };
        }
```